### ISSUE TITLE
Bug: [no-base-to-string] ignoredTypeNames should match on the type name without generics

### ISSUE THREAD

**Issue by tyilo** [typescript-eslint/typescript-eslint#11521]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.ts&code=JYOwLgpgTgZghgYwgAgLIE8CiUoHsoA8AKgHzIDeyAvgFA0AmECANnFCgriAM5jLR4oALjRYc%2BAiACuAWwBG0EgG4aA-ADowuAMpgooAOYAKAJRKgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oCMBDZRLXxdk%2BKkwDm6MAG1w2HImjQO0SABp5CrAuyRKEztEQATAColEAOV4BbFNLm7dkALKEAostWQtugLp%2BEAC%2BfoEhIMFAA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
interface MyError<T> { }

declare const error: MyError<number>;
error.toString();
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-base-to-string": [
      "error",
      {
        "ignoredTypeNames": [
          "MyError"
        ]
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

There should be no lints.

### Actual Result

```
'error' will use Object's default stringification format ('[object Object]') when stringified. 4:1 - 4:6
```

### Additional Info

I know you could add `MyError<number>` to the `ignoredTypeNames`, but that would only take care of a single possible generic type.

At runtime the generics don't exist, so whether `toString()` is overridden or not, can't depend on the generics. I think the `ignoredTypeNames` should be based on the type without generics.

There is some discussion in https://github.com/typescript-eslint/typescript-eslint/issues/11420

**Comment by JoshuaKGoldberg** [comment]
Agreed on this one too, users can't be expected to write out all the variations of `<>`s.
